modules/aggregation-layers/src/heatmap-layer/heatmap-layer.js

```diff
@@ -496,17 +501,18 @@ export default class HeatmapLayer extends AggregationLayer {
 
   _debouncedUpdateWeightmap(fromTimer = false) {
     let {updateTimer} = this.state;
+    const {debounceTimeout} = this.props;
 
     if (fromTimer) {
       updateTimer = null;
       // update
       this._updateBounds(true);
       this._updateTextureRenderingBounds();
-      this.setState({isWeightMapDirty: true});
+      this._updateWeightmap();
```

Comment:
I suggest that you keep the old code if there's no particularly good reason to make this change. Using a dirty flag in layer state instead of calling update immediately ensures that update is run at most once per render cycle.